### ISSUE TITLE
PRTG: staging.sabilling.com HTTPS 502 Bad Gateway

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1945]
In GitLab by @kdjstudios on Oct 21, 2020, 08:03

**HD:** https://www-servicedesk.answernet.com/profiles/ticket/2020-10-20-34443/conversation

The staging server is not running correctly.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 21, 2020, 08:06

@aznbyte @Robleski 

After confirming that no one was using the server and given the okay to re-pull the production git and restart SA Billing, I proceeded to do so, but was meet with the following error after trying to restart SA Billing. 

![image](/uploads/543762d6454f62449c055bfbe0b9a8e4/image.png)

I performed the bundle install and it shows that the gem is installed. But somehow it looks like SA Billing can not find it.

![image](/uploads/6390a2eb3039b02b6340d8389e56ccdf/image.png)

I am not familiar with this error so will most likely need some assistance. 

EDIT:

After some more research online I found that deleting the gemlock file and then trying the bundle install will "force" a reinstall and should fix this. I performed that and got errors about missing SSL certs, which after looking online found that those are due to using "https" in the gemfile. So I removed those and am now processing the bundle install.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 21, 2020, 10:24

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 21, 2020, 11:59

@aznbyte @Robleski 

Here is a current status of each dev server and the steps I performed. I logged in via the `sabilling` user on each. Jordan I feel we ran into similar permission issues on production servers this last time too, but we were able to override them at the time. I suggest we also check the production servers too, but not during business hours. 

NOTE: I also went looking for the instructions to update/fix permissions, however I can not find them. @aznbyte did you document those steps last time you worked with David? If not, may you please this time, so we can reference them in the future. 

## [Not working] dev.sabilling.com
```
cd /opt/sabilling
git checkout master
git pull
rake assets:precompile RAILS_ENV=production --trace
sudo service sabilling restart

```
![image](/uploads/34466ee900558d0ed32b9071fe4df6da/image.png)

## [Not working] dev2.sabilling.com
```
cd /opt/sabilling
git checkout master
git pull
rake assets:precompile RAILS_ENV=production --trace

```
![image](/uploads/c593650195eb6c33ea27cabf2cfa37e7/image.png)

## [Not working] merge.sabilling.com
```
cd /opt/sabilling
git checkout master
git pull
rake assets:precompile RAILS_ENV=production --trace

```
![image](/uploads/e0c5936baec0cc5f4c4c296c897b6395/image.png)

## [Working] merge2.sabilling.com
```
cd /opt/sabilling
git checkout master
git pull
rake assets:precompile RAILS_ENV=production --trace
sudo service sabilling restart

```
![image](/uploads/1db6c7a4a7b99cebaeec6799010feb0a/image.png)

## [Not working]  staging.sabilling.com
```
cd /opt/sabilling
git checkout master
git pull
```
![image](/uploads/4a3a312cde53fc152bd7c7a97aaf74b2/image.png)

performing the following seems to "force" git to pull.

```
git fetch --all
git reset --hard origin/master
```

Then attempting the following results in error

```
rake assets:precompile RAILS_ENV=production --trace
sudo service sabilling restart

```
![image](/uploads/cfab7bc4781a7195d136e937c33ba138/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 21, 2020, 11:59

assigned to @kdjstudios

**Comment by kdjstudios** [comment]
In GitLab by @aznbyte on Oct 21, 2020, 14:43

On staging, I was able to get a little further but the current issue is this:

![image](/uploads/acbcd3bad872f6096144922409c97add/image.png)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 28, 2020, 07:48

@Robleski @aznbyte 

I was able to get all but the dev2 server up and working. I changed permission on a few directories and had to do a few bundle installs, but it appears all the other servers are working as expected. I created a script in the jobs dir on each install that will pull the master sabilling and then compile and restart all services.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 11, 2020, 13:33

I was able to work with Ben today on the issues with the dev2 server's connection to LDAP and we manually forced the IP address via updating the ldap.yml file to point directly to the IP rather then to "LDAP_IP". we so far have been unable to determine where in the system that environment variable is being set, but that is the cause of the issues. Marking this into testing for now.
